File: code/c#/src/misc/matrix.cs
using System;

public static class Matrix
{
	public static int findPath(int [][] mat)
	{
		for (int r = 0; r < mat.Length; r++)
		{
			int [] cols = mat[r];
			for (int c = 0; c < cols.Length; c++)
			{
				if (mat[r,c] == 1)
					Console.WriteLine(r,c);
			}
		}
	}

	public static void Main()
	{
		int [][] mat = new int [][] {new int[] {0,0}, new int[] {1, 0}, 
			new int[]{0,0}, new int[]{0,0}, new int[]{0,0}, new int[]{0,0},
			new int[]{0,1}, new int[]{0,0}};
		Console.WriteLine(findPath(mat));
	}
}
	

File: code/math/linearAlgebra/src/Matrix.py

class Matrix:

    def __init__(self, mat):
        self.mat = mat

    def gaussian(self, row1, row2, scalar):
        
        for i in range(len(self.mat[row1])):
            self.mat[row2][i] += self.mat[row1][i] * scalar 

    def reduced_echelon_form(self):
        pass

    def row_reduced_echelon_form(self):
        pass



    


